Source/Core/VideoCommon/VertexShaderGen.cpp

```diff
@@ -461,6 +461,22 @@ ShaderCode GenerateVertexShaderCode(APIType api_type, const vertex_shader_uid_da
   // get rasterized correctly.
   out.Write("o.pos.xy = o.pos.xy - o.pos.w * " I_PIXELCENTERCORRECTION ".xy;\n");
 
+  // By now our position is in clip space
+  // however, higher resolutions than the Wii outputs
+  // cause an additional pixel offset
+  // due to a higher pixel density
+  // we need to correc this by converting our
+  // clip-space position into the Wii's screen-space
+  // acquire the right pixel and then convert it back
+  out.Write("float ss_pixel_x = ((o.pos.x + 1.0f) * (640.0f * 0.5f));\n");
+  out.Write("float ss_pixel_y = ((o.pos.y + 1.0f) * (480.0f * 0.5f));\n");
```

Comment:
Yup, didn't expect this to be good out of the gate honestly.  Just posting what I had!

I didn't think about the viewport thing.  Is there a way to access the current viewport in shader?